x-pack/legacy/plugins/fleet/public/pages/agent_list/components/agent_enrollment.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, Fragment } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiSpacer,
+  EuiText,
+  EuiFilterGroup,
+  EuiFilterButton,
+} from '@elastic/eui';
+import { FrontendLibs } from '../../../lib/types';
+import {
+  ShellEnrollmentInstructions,
+  ContainerEnrollmentInstructions,
+  ToolsEnrollmentInstructions,
+} from './enrollment_instructions';
+
+interface RouterProps {
+  libs: FrontendLibs;
+  onClose: () => void;
+}
+
+export const AgentEnrollmentFlyout: React.SFC<RouterProps> = ({ libs, onClose }) => {
+  const [quickInstallType, setQuickInstallType] = useState<'shell' | 'container' | 'tools'>(
+    'shell'
+  );
+
+  const renderHeader = () => (
```

Comment:
small suggestion nothing blocking I like to avoid to create function when I can why not directly assign here

```jsx
const header = (
  <EuiFlyoutHeader>
  </EuiFlyoutHeader>
) ;
```